scrapy/core/downloader/handlers/http11.py

```diff
@@ -285,6 +286,12 @@ def _get_agent(self, request, timeout):
             scheme = _parse(request.url)[0]
             proxyHost = to_unicode(proxyHost)
             omitConnectTunnel = b'noconnect' in proxyParams
+            if omitConnectTunnel:
+                warnings.warn("Using HTTPS proxies in the noconnect mode is deprecated. "
+                              "If you use Crawlera, it doesn't require this mode anymore, "
+                              "so you should update scrapy-crawlera to 1.3.0+ "
+                              "and remove '?noconnect' from the Crawlera URL.",
+                              ScrapyDeprecationWarning)
```

Comment:
I’m guessing we need to use a higher stack level to point to the user code where noconnect is used.

Reply:
I'm afraid it's not possible as there is a lot of unrelated frames there, also the user code is usually in `settings.py`, in `CRAWLERA_URL`.